### PR TITLE
Improve error message for invalid example pipeline value

### PR DESCRIPTION
**Pull request by merelcht**
## Description
I noticed that the error message on entering an invalid example pipeline value wasn't very descriptive:

Screenshot 2024-01-22 at 16 19 37

## Development notes
Updated the message to repeat the entered value to the user and mention that the value was invalid specifically for the example pipeline flag:

Screenshot 2024-01-22 at 16 19 14

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: tests/framework/cli/test_starters.py [hunk 1]
```diff
@@ -1311,7 +1312,8 @@ def test_invalid_example(self, fake_kedro_cli, bad_input):
 
         assert result.exit_code != 0
         assert (
-            "It must contain only y, n, YES, NO, case insensitive.\n" in result.output
+            "It must contain only y, n, YES, or NO (case insensitive).\n"
```
Comment (AhdraMeraliQB) [inline tests/framework/cli/test_starters.py:1315]: Should we test for the full expected error message here? Like we do for the test above

```
assert f"{bad_input} is an invalid value for example pipeline." in result.output`
```

**Review by AhdraMeraliQB — COMMENTED** [review]
Looks good to me - just a small comment on one of the tests. We should probably add a test to `TestToolsAndExampleFromCLI` for an invalid `--example` input

File: tests/framework/cli/test_starters.py (hunk 1 above)
Comment (merelcht) [reply tests/framework/cli/test_starters.py:1320]: Done!

**Comment by merelcht** [comment]
> Looks good to me - just a small comment on one of the tests. We should probably add a test to `TestToolsAndExampleFromCLI` for an invalid `--example` input

This is a good point. However, looking at `test_starters.py` I feel like the full file can use some refactoring. It contains far too many tests IMO and it's really hard to figure out what test should be where. E.g. without reading all tests I don't understand the difference between the classes `TestToolsAndExampleFromUserPrompts` and `TestNewFromUserPromptsValid`. There's also inconsistency in the invalid/valid format. So rather than adding more tests now I think the test file must be refactored.

**Review by AhdraMeraliQB — APPROVED** [review]
LGTM - we can add that test case in/after the refactor!

**Review by lrcouto — APPROVED** [review]

**Comment by lrcouto** [comment]
> > Looks good to me - just a small comment on one of the tests. We should probably add a test to `TestToolsAndExampleFromCLI` for an invalid `--example` input
> 
> This is a good point. However, looking at `test_starters.py` I feel like the full file can use some refactoring. It contains far too many tests IMO and it's really hard to figure out what test should be where. E.g. without reading all tests I don't understand the difference between the classes `TestToolsAndExampleFromUserPrompts` and `TestNewFromUserPromptsValid`. There's also inconsistency in the invalid/valid format. So rather than adding more tests now I think the test file must be refactored.

Maybe we could separate the tools-related tests to their own file, since there's a decent amount of them.

**Comment by merelcht** [comment]
> Maybe we could separate the tools-related tests to their own file, since there's a decent amount of them.

I've been trying some things here: https://github.com/kedro-org/kedro/pull/3550 
But separating them isn't as easy as I thought.. It creates some issues with the `cookiecutter` cache. I think the main question is how we properly test the use cases where the real starters need to be accessed.

**Pull request merged**